app/services/log_event_searcher.rb

```diff
@@ -74,17 +74,22 @@ def filter_statements
     statements = Statement.all
 
     if invoice_number.present?
-      statements = statements.where(
-        "CAST(statements.id AS CHAR) LIKE ? OR CONCAT(account_id, '-', id) LIKE ?", "%#{invoice_number}%", "%#{invoice_number}%"
-      )
+      statements = if Nucore::Database.oracle?
+                     statements.where(
+                       "TO_CHAR(statements.id) LIKE ? OR account_id || '-' || id LIKE ?", "%#{invoice_number}%", "%#{invoice_number}%"
+                     )
+                   else
+                     statements.where(
+                       "CAST(statements.id AS CHAR) LIKE ? OR CONCAT(account_id, '-', id) LIKE ?", "%#{invoice_number}%", "%#{invoice_number}%"
+                     )
+                   end
```

Comment:
nit: can we only set the query in the if-else and have only one `where(..., "%#{invoice_number}%", "%#{invoice_number}%")`